### DISCUSSION TITLE
Optionally disabling NAT-T

### DISCUSSION THREAD

**Discussion by pauldraper** [strongswan/strongswan#653] [Ideas]
Continuation of discussion https://wiki.strongswan.org/issues/1265

> I'd like to ask if it would be possible to add an option to disable NAT-T floating to port UDP 4500.

In my case, I am using AWS EC2, for which even for "public" instances still use a 1:1 NAT. However, the NAT permits ESP packets. 

I believe(?) this feature already exists, by disabling MOBIKE, which is the reason NAT traversal happens.

```
mobike = no
```

**Comment by Thermi** [comment]
No, no, and no

Nat traversal is an integral part of IKEv2 and you can't disable it. MOBIKE just makes the peers float to 4500 regardless of NAT.

Am 4. Oktober 2021 15:57:07 UTC schrieb Paul Draper ***@***.***>:
>Continuation of discussion https://wiki.strongswan.org/issues/1265
>
>> I'd like to ask if it would be possible to add an option to disable NAT-T floating to port UDP 4500.
>
>In my case, I am using AWS EC2, which even for "public" instances, still uses a 1:1 NAT. Encapsulated ESP packets is unnecessary. 
>
>I believe(?) this feature already exists, by disabling MOBIKE, which is the reason NAT traversal happens.
>
>```
>mobike = no
>```
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/653
-- 
Sent from mobile

**Comment by tobiasbrunner** [comment]
While it's true that NAT-T is an integral part of IKEv2 (i.e. it's not a separate draft/RFC as it was with IKEv1), the feature as such is optional (RFC 7296, section 2.23 explicitly states: "Support for NAT traversal is optional."). So it would theoretically be possible to add an option to disable NAT-T for a connection.

But doing so on a server with static NAT mapping would break the connection if the client was behind a dynamic NAT or actually required UDP encap and sends some random NAT-D payload. So maybe instead of disabling NAT-T outright, the result of the NAT detection based on the initiator's NAT-D payloads should be considered (i.e. only omit the NAT-D payloads in the response if the client does not appear to be behind a NAT). Enabling the option as initiator could only be done safely if it's known beforehand that both the initiator and the responder are not behind a NAT or it's a static mapping.

**Comment by marek22k** [comment]
What is the current status of the feature request?
I would also like to be able to disable NAT-T for server-to-server VPNs. If one of the servers is misconfigured and is behind a NAT, the connection should fail and not continue over UDP. Among other reasons, this is because it could change the MTU and thus lead to fragmentation.
